packages/repository/src/relations/has-and-belongs-to-many/has-and-belongs-to-many.helpers.ts

```diff
@@ -0,0 +1,205 @@
+import debugFactory from 'debug';
+import {camelCase} from 'lodash';
+import {
+  DataObject,
+  deduplicate,
+  DeepRequired,
+  Entity,
+  HasAndBelongsToManyDefinition,
+  InvalidRelationError,
+  isTypeResolver,
+  RelationType,
+} from '../..';
+
+const debug = debugFactory(
+  'loopback:repository:relations:has-and-belongs-to-many:helpers',
+);
+
+export type HasAndBelongsToManyResolvedDefinition = DeepRequired<
+  HasAndBelongsToManyDefinition
+>;
+
+export function resolveHasAndBelongsToManyMetadata(
```

Comment:
I notice that the code is the same/similar to [`has-many-through.helpers.ts`](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/relations/has-many/has-many-through.helpers.ts). As these two relations are similar, maybe we can reuse the code to simplify it.

Reply:
Yes, much of the code I use is the same as in HasManyThrough as I explain in the description of the PR. I will work to simplify it.